netfx_loader/ClrLoader.cs

```diff
@@ -29,11 +38,9 @@ public static class ClrLoader
 
                 Print($"Located domain {domain}");
 
-                var handle = GCHandle.Alloc(domain, GCHandleType.Pinned);
-
-                Print($"Created handle {handle}");
-
-                return handle.AddrOfPinnedObject();
+                var domainData = new DomainData(domain);
+                _domains.Add(domainData);
+                return new IntPtr(_domains.Count - 1);
```

Comment:
Honestly, would it be a bit complex for it? I just said `GCHandleType.Pinned` is not necessary, `Normal` for `DomainData` just fine.😂 Well, but this method indeed would be safer because passing an invalid GCHandle.FromIntPtr may crash the application.

Reply:
Yeah, it got a bit out of hand to fix the ownership issues.